driver/reduce_driver.hpp

```diff
@@ -493,20 +493,20 @@ int ReduceDriver<Tgpu, Tref>::VerifyForward()
     if(std::is_same<Tgpu, float>::value && reduceOp == MIOPEN_REDUCE_TENSOR_NORM2)
         tolerance *= 12.0;
 
-    if(error > tolerance)
+    if(!std::isfinite(error) || error > tolerance)
     {
-        std::cout << "ReduceTensor() Failed with error = " << error
-                  << " , tolerance = " << tolerance << "\n";
+        std::cout << "ReduceTensor() FAILED with error = " << error
+                  << " , tolerance = " << tolerance << std::endl;
     }
     else
     {
         if(out_indices.size() > 0)
         {
             auto error2 = miopen::rms_range(outhost_indices, out_indices);
 
-            if(static_cast<float>(error2) != 0.0f)
```

Comment:
@qianfengz and @atamazov  please check if this change meets your design requirements.